photon-api/src/main/scala/com/linkedin/photon/ml/evaluation/MultiEvaluator.scala

```diff
@@ -55,11 +55,44 @@ abstract class MultiEvaluator(
     // TODO: Log which IDs had invalid evaluation metric values
     scoresAndLabelsAndWeights
       .join(ids)
-      .map { case (uniqueId, (scoreLabelAndWeight, id)) => (id, scoreLabelAndWeight) }
+      .map { case (_, (scoreLabelAndWeight, id)) => (id, scoreLabelAndWeight) }
       .groupByKey()
       .values
       .map(scoreLabelAndWeights => localEvaluator.evaluate(scoreLabelAndWeights.toArray))
       .filter(result => !(result.isInfinite || result.isNaN))
       .mean()
   }
+
+  /**
+   * Method used to define equality on multiple class levels while conforming to equality contract. Defines under
+   * what circumstances this class can equal another class.
+   *
+   * @param other Some other object
+   * @return Whether this object can equal the other object
+   */
+  override def canEqual(other: Any): Boolean = other.isInstanceOf[MultiEvaluator]
+
+  /**
+   * Compares two [[MultiEvaluator]] objects.
+   *
+   * @param other Some other object
+   * @return True if the both models conform to the equality contract and have the same model coefficients, false
+   *         otherwise
+   */
+  override def equals(other: Any): Boolean = other match {
+    case that: MultiEvaluator =>
+      that.canEqual(this) &&
+        (this.localEvaluator == that.localEvaluator) &&
+        (this.ids == that.ids) &&
+        super.equals(that)
+
+    case _ => false
+  }
+
+  /**
+   * Returns a hash code value for the object.
+   *
+   * @return An [[Int]] hash code
+   */
+  override def hashCode: Int = (41 * (41 * super.hashCode()) + localEvaluator.hashCode()) + ids.hashCode()
```

Comment:
FYI -- There's a Java builtin for building composite hash codes now:

```java
    override def hashCode: Int = Objects.hash(super, localEvaluator, ids)
```

Though it's probably just going to do what you've done here.